test/suite/test_rollback_to_stable26.py

```diff
@@ -0,0 +1,137 @@
+        s.rollback_transaction()
+        evict_cursor.close()
+
+    def test_rollback_to_stable26(self):
+        # Create a table without logging.
+        uri = "table:rollback_to_stable26"
+        format = 'key_format={},value_format=S'.format(self.key_format)
+        self.session.create(uri, format + ', log=(enabled=false)')
+
+        # Pin oldest timestamp to 10.
+        self.conn.set_timestamp('oldest_timestamp=' + self.timestamp_str(10))
+
+        # Start stable timestamp at 10.
+        self.conn.set_timestamp('stable_timestamp=' + self.timestamp_str(10))
+
+        # I've chosen key_b such that if the test fails it takes about fifteen seconds on an
+        # old machine. This gives a margin against faster machines accidentally accepting the
+        # failing behavior without taking excessively long if the failure behavior does appear.
+        # Increase if necessary in the future.
+        key_a = 1
+        #key_b = 500000000
+        key_b = 80000000
+        key_c = key_b * 2
+
+        value_a = "aaaaa" * 100
+        value_b = "bbbbb" * 100
+        value_c = "ccccc" * 100
+
+        s = self.conn.open_session()
+        cursor = s.open_cursor(uri)
+
+        # If requested, write key_b at time 25.
+        if self.btime == 25:
+             s.begin_transaction()
+             cursor[key_b] = value_b
+             s.commit_transaction('commit_timestamp=' + self.timestamp_str(25))
+
+        # Write the other keys at time 50.
+        s.begin_transaction()
+        cursor[key_a] = value_a
+        cursor[key_c] = value_c
+        s.commit_transaction('commit_timestamp=' + self.timestamp_str(50))
+
+        # If requested, evict now.
+        if self.evicttime == 50:
+            self.evict(s, uri, key_a, value_a)
+
+        # If requested, write key_b at time 75.
+        if self.btime == 75:
+             s.begin_transaction()
+             cursor[key_b] = value_b
+             s.commit_transaction('commit_timestamp=' + self.timestamp_str(75))
+
+        # If requested, evict now.
+        if self.evicttime == 75:
+            self.evict(s, uri, key_a, value_a)
+
+        cursor.close()
+
+        # Now roll back to 60. Allow it one second. It should take far less than that. If it
+        # tries to process the intermediate keys (implausible for rows, used to happen in some
+        # cases for columns) it'll take considerably longer. Python doesn't guarantee that the
+        # timer resolution is any better than seconds, so we have to accept "1" second as a
+        # possible manifestation of a much shorter time.
+        #
+        # Note that while it would be better to have the test time out instead of waiting for
+        # the operation to finish, there doesn't appear to be any good way to do that without
+        # using Unix interfaces and making the test non-portable.
+        self.conn.set_timestamp('stable_timestamp=' + self.timestamp_str(60))
+        start = time.time()
```

Comment:
Use time.time_ns()?

Reply:
The Python manual isn't clear on whether time_ns() is any better, it just says "similar to time()" which warns that some systems don't provide better that one-second time resolution.

I'm also not sure whether this is a real issue or reflects Python cargo-culting of autoconf tests for Unix time interfaces that stabilized 20+ years ago and the only systems on which it won't work are ancient SVR4 clones nobody cares about. Or what the situation on Windows is. My inclination is to be conservative about these things.

If you think it's likely ok I can change it and set the threshold to say 0.5 seconds.

Reply:
(no longer relevant, code's been removed)